src/sqlfluff/core/parser/segments/raw.py

```diff
@@ -41,8 +41,10 @@ def __init__(
         else:
             self._raw = self._default_raw
         self._raw_upper = self._raw.upper()
-        # pos marker is required here
-        self.pos_marker = pos_marker
+        # pos marker is required here. We ignore the typing initially
+        # because it might *initially* be unset, but it will be reset
```

Comment:
I do mean that, but I tried having it a `Optional[PositionMarker]` and that makes a load of other things downstream complain a lot. It seemed easier to go down this route than change a lot of downstream code. I'll have a look at how possible that would be to resolve. 🤔

Reply:
Yeah - I've just tried re-enabling that, and it has downstream issues, because for `BaseSegment` it really isn't optional and a lot of the code is written there assuming that it isn't. I think the hacks it would take there aren't worth the clarity we gain here...

Reply:
No worries! This was more of a nitpick. I imagine this can be a similar experience as trying to make C++ code 100% "const correct", which I have suffered through in the past. 😓